tour_model_eval/viz_bin_cluster.py

```diff
@@ -0,0 +1,122 @@
+                 [curr_trip.data.end_loc["coordinates"][1], curr_trip.data.end_loc["coordinates"][0]]], weight=2,
+                color=color_map(index))
+            layer.add_to(map)
+    map.add_child(color_map)
+    return map
+
+
+# def cutoff_bins_map(bins, trips):
+#     color_map = cm.linear.Set1_07.to_step(len(bins), index=[i for i in range(len(bins))])
+#     map = folium.Map(
+#         location=[trips[0].data.start_loc["coordinates"][1], trips[0].data.start_loc["coordinates"][0]],
+#         zoom_start=12, max_zoom=30, control_scale=True)
+#     for t, bin in enumerate(bins):
+#         for trip_index in bin:
+#             # curr_trip = bin_trips[curr_trip_index]
+#             curr_trip = trips[trip_index]
+#             layer = folium.PolyLine(
+#                 [[curr_trip.data.start_loc["coordinates"][1], curr_trip.data.start_loc["coordinates"][0]],
+#                  [curr_trip.data.end_loc["coordinates"][1], curr_trip.data.end_loc["coordinates"][0]]], weight=2,
+#                 color=color_map(t))
+#             layer.add_to(map)
+#     map.add_child(color_map)
+#     return map
+
+
+def clusters_map(labels, points, clusters):
+    # Each color represents a label
+    # labels have to be in order in the colormap index
+    labels_clt = list(set(sorted(labels)))
+    color_map = cm.linear.Set1_07.to_step(clusters, index=[i for i in labels_clt])
+
+    # Plot all clusters with different colors on the map
+    # Choose the first start point to locate the map
+    map = folium.Map(location=[points[0][1], points[0][0]], zoom_start=12, max_zoom=30, control_scale=True)
+
+    if labels:
+        for i, point in enumerate(points):
+            start_lat = point[1]
+            start_lon = point[0]
+            end_lat = point[3]
+            end_lon = point[2]
+            layer = folium.PolyLine([[start_lat, start_lon],
+                                     [end_lat, end_lon]], weight=2, color=color_map(labels[i]))
+            layer.add_to(map)
+    map.add_child(color_map)
+    return map
+
+
+def specific_bin_map(bins, trips):
+    color_map = cm.linear.Set1_07.to_step(len(bins), index=[i for i in range(len(bins))])
+    # Plot trips in the same bin,
+    map = folium.Map(
+        location=[trips[0].data.start_loc["coordinates"][1], trips[0].data.start_loc["coordinates"][0]],
+        zoom_start=12, max_zoom=30, control_scale=True)
+
+    # Here we to choose a specific bin
+    for t in range(0, 3):
+        for i in range(len(bins[t])):
+            curr_trip = trips[bins[t][i]]
+            layer = folium.PolyLine(
+                [[curr_trip.data.start_loc["coordinates"][1], curr_trip.data.start_loc["coordinates"][0]],
+                 [curr_trip.data.end_loc["coordinates"][1], curr_trip.data.end_loc["coordinates"][0]]], weight=2,
+                color=color_map(t))
+
+            layer.add_to(map)
+    map.add_child(color_map)
+    return map
+
+
+def specific_cluster_map(labels, points, clusters):
```

Comment:
isn't this just a duplicate of `clusters_map` with a different set of clusters passed in?
Again, can merge.
DRY.

Reply:
They look like similar. But I add a if statement in this function. If I merge them, then other people might need to fix the indentation every time and they couldn't run all at a time. I don't know how to avoid this situation .